components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/IdentityProvider.java

```diff
@@ -365,19 +384,182 @@ public void setAlias(String alias) {
+                } catch (JSONException e) {
+                    // Handle plain text certificate for file based configuration.
+                    if (certificateValue.startsWith(IdentityUtil.PEM_BEGIN_CERTFICATE)) {
+                        this.certificateInfoArray = handlePlainTextCertificate(certificateValue);
+                    } else {
+                        // Handle encoded certificate values. While uploading through UI.
+                        this.certificateInfoArray = handleEncodedCertificate(certificateValue);
+                    }
+                }
+            }
+        } catch (NoSuchAlgorithmException e) {
+            log.error("Error while generating thumbPrint. Unsupported hash algorithm " + e);
+        }
+    }
+
+    /**
+     * handle the certificates which is in json format.
+     * @param certificateValue
+     * @return array of certificate value and thumbPrint of each certificates.
+     * @throws NoSuchAlgorithmException
+     */
+    private CertificateInfo[] handleJsonFormatCertificate(String certificateValue) throws NoSuchAlgorithmException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Handling json format certificate.");
+        }
+        JSONArray jsonCertificateInfoArray = new JSONArray(certificateValue);
+        if (jsonCertificateInfoArray.length() > 1 && log.isDebugEnabled()) {
+            log.debug("More than one certificate has been found");
+        }
+        List<CertificateInfo> certificateInfos = new ArrayList<>();
+        for (int i = 0; i < jsonCertificateInfoArray.length(); i++) {
+            JSONObject jsonCertificateInfoObject = (JSONObject) jsonCertificateInfoArray.get(i);
+
+            CertificateInfo certificateInfo = new CertificateInfo();
+            certificateInfo.setThumbPrint(IdentityApplicationManagementUtil.generateThumbPrint(
+                    jsonCertificateInfoObject.getString(THUMB_PRINT)));
+            certificateInfo.setCertValue(jsonCertificateInfoObject.getString(CERT_VALUE));
+            certificateInfos.add(certificateInfo);
+        }
+        return certificateInfos.toArray(new CertificateInfo[jsonCertificateInfoArray.length()]);
+    }
+
+    /**
+     * handle the certificate which is in encoded format.
+     *
+     * @param certificateValue
+     * @return array of certificate value and thumbPrint of each certificates.
+     * @throws NoSuchAlgorithmException
+     */
+    private CertificateInfo[] handleEncodedCertificate(String certificateValue) throws NoSuchAlgorithmException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Handling encoded certificates.");
+        }
+        String decodedCertificate = new String(Base64.getDecoder().decode(certificateValue));
+        int numberOfCertificates = StringUtils.countMatches(decodedCertificate,
+                IdentityUtil.PEM_BEGIN_CERTFICATE);
+
+        if (numberOfCertificates > 1) {
```

Comment:
What if the numberOfCertificates are 0? 
We can return without proceeding, can't we?

Reply:
If numberOfCertificates zero then for loop won't execute since it is starting from 1